blueprints/in-repo-engine/index.js

```diff
@@ -32,14 +34,31 @@ module.exports = {
     }
   ],
 
+  _setupRouteBlueprint: function() {
+    var checker = new VersionChecker(this);
+    var emberCliVersion = checker.for('ember-cli', 'npm');
+    var routeBlueprintPkg = emberCliVersion.isAbove('2.6.0')
+      ? OLD_ROUTE_BLUEPRINT_PKG
+      : NEW_ROUTE_BLUEPRINT_PKG;
+
+    this.routeBlueprint = require(routeBlueprintPkg + '/blueprints/route/index');
+
+    this.shouldTouchRouter = this.routeBlueprint.shouldTouchRouter;
+    this.shouldEntityTouchRouter = this.routeBlueprint.shouldEntityTouchRouter;
+  },
+
   install: function(options) {
     this.options = options;
-    return this._super.install.apply(this, arguments);
+    var superResult = this._super.install.apply(this, arguments);
+    this._setupRouteBlueprint();
+    return superResult;
```

Comment:
This approach is a little awkward, but I see no other way given that:
* We have to consider that super's install hook may return a promise, and so we must pass that through
* `this._super.install` (and uninstall) have side effects that `ember-cli-version-checker` depend on (specifically, the existence of the `project` attribute, from which the npm dependency data is fetched)